### PR TITLE
add like app

### PR DESCRIPTION
**Pull request by black-art**
## TRELLO

https://trello.com/c/tQvHsCxc/27-create-like-app

## Code reviewers

- [x] @hrytskivr
- [ ] @spycheast
- [x] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/like/views.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class LikeView(View):
+    """Likes view handles GET, POST, DELETE requests."""
+
+    def get(self, request, like_id):
+        """
+        Handles GET request, that return JSON response with HTTP status 200,
+        if exception: HTTP status 404.
+        """
+        like = Like.get_by_id(like_id)
+        print(like_id)
```
Comment (oshtohryn) [inline myTrip/like/views.py]: remove print

File: myTrip/like/views.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        if not like:
+            return HttpResponse(status=404)
+        like = like.to_dict()
+        return JsonResponse(like, status=200)
```
Comment (Ashiato) [inline myTrip/like/views.py]: status  201 Created

File: myTrip/like/views.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    def post(self, request):
+        """Handles POST request, that return JSON response with HTTP status 200."""
+        like_data = json.loads(request.body.decode('utf-8'))
+        like = Like.create(**like_data)
+        return JsonResponse(like.to_dict(), status=200)
```
Comment (oshtohryn) [inline myTrip/like/views.py]: status 201 Created

File: myTrip/like/models.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
+"""This module contains like model class and basic functions."""
```
Comment (hrytskivr) [inline myTrip/like/models.py]: Suggestion: This module contains Like model and basic methods.

File: myTrip/like/models.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    user = models.IntegerField()
+    trip = models.IntegerField(null=True)
+    checkpoint = models.IntegerField(null=True)
+    photo = models.IntegerField(null=True)
+    comment = models.IntegerField(null=True)
+
+    @staticmethod
+    def get_by_id(like_id):
+        """
+        ToDo method to get one element by his id, uses to views,
```
Comment (hrytskivr) [inline myTrip/like/models.py]: Remove 'ToDo's from everywhere, when pushing to remote repo.
Gonna be like this: A method to get specific element by id. Used in views. Returns None when exception works.

**Review by hrytskivr — APPROVED** [review]
Nice1

File: myTrip/like/apps.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class LikeConfig(AppConfig):
+    """."""
```
Comment (lhalam) [inline myTrip/like/apps.py]: delete this line

File: myTrip/like/models.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    """
+    Like
+    :argument id: int - auto generate primary key
+    :argument user: int - ToDo foreign key
+    :argument trip: int - ToDo foreign key
+    :argument checkpoint: int - ToDo foreign key
+    :argument photo: int - ToDo foreign key
+    :argument comment: int - ToDo foreign key
+    """
+    user = models.IntegerField()
```
Comment (lhalam) [inline myTrip/like/models.py]: add the empty line

File: myTrip/like/models.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        try:
+            return Like.objects.get(id=like_id)  # may be changed to method filter
+        except ObjectDoesNotExist:
+            return None
+
+    @staticmethod
+    def get_by_trip_id(trip_id):
+        """
+        A method to get one like by trip id.
```
Comment (lhalam) [inline myTrip/like/models.py]: for example:
```
"""
Get Like with given trip id
 Args:
    trip_id (int): trip id.
Returns:
    QuerySet<Like>: QuerySet of Like.
"""
```

File: myTrip/like/models.py [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        Used in views.
+        Returns None when exception works.
+        """
+        try:
+            return Like.objects.get(id=comment_id)  # may be changed to method filter
+        except ObjectDoesNotExist:
+            return None
+
+    def to_dict(self):
+        """A method rebuilds queryset to object dictionary for our views."""
```
Comment (lhalam) [inline myTrip/like/models.py]: for example:
```
"""Convert model object to dictionary.
        Return:
            dict:
                {
                    'id': id,
                    'user_id': name,
                    ...
                }
        """
```

File: myTrip/like/models.py [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        """A method delete like by id."""
+        try:
+            like = Like.objects.get(id=like_id)
+            like.delete()
+            return True
+        except ObjectDoesNotExist:
+            return None
+
+    def __str__(self):
+        return "like={} user={} trip={} checkpoint={} photo={} comment={}".format(self.id,
```
Comment (lhalam) [inline myTrip/like/models.py]: like={} => id={}

File: myTrip/like/urls.py [hunk 11]
```diff
@@ -0,0 +1,7 @@
+"""."""
```
Comment (lhalam) [inline myTrip/like/urls.py]: delete this line

**Pull request merged**